### ISSUE TITLE
fs:lustre component fails to compile on several systems with Lustre present

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#12181]
It appears that commit   c6a6c256e3687459a3c58bf31bff413a421b58a5  introduced a regression and the lustre fs component no longer builds on at least NERSC perlmutter system:

```
ake[2]: Entering directory '/global/u2/h/hpp/ompi/ompi/mca/fs/lustre'
depbase=`echo fs_lustre_file_open.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../../../opal/include -I../../../../ompi/include -I../../../../oshmem/include -I../../../../ompi/mpiext/cuda/c -I../../../../ompi/mpiext/rocm/c   -iquote../../../..  -I/global/homes/h/hpp/ompi/3rd-party/openpmix/include -I/global/homes/h/hpp/ompi/3rd-party/openpmix/include -I/global/homes/h/hpp/ompi/3rd-party/openpmix/ -I/global/homes/h/hpp/ompi/3rd-party/openpmix/  -O3 -DNDEBUG  -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wformat-truncation=0 -finline-functions -mcx16 -MT fs_lustre_file_open.lo -MD -MP -MF $depbase.Tpo -c -o fs_lustre_file_open.lo fs_lustre_file_open.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../../../opal/include -I../../../../ompi/include -I../../../../oshmem/include -I../../../../ompi/mpiext/cuda/c -I../../../../ompi/mpiext/rocm/c -iquote../../../.. -I/global/homes/h/hpp/ompi/3rd-party/openpmix/include -I/global/homes/h/hpp/ompi/3rd-party/openpmix/include -I/global/homes/h/hpp/ompi/3rd-party/openpmix/ -I/global/homes/h/hpp/ompi/3rd-party/openpmix/ -O3 -DNDEBUG -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wformat-truncation=0 -finline-functions -mcx16 -MT fs_lustre_file_open.lo -MD -MP -MF .deps/fs_lustre_file_open.Tpo -c fs_lustre_file_open.c  -fPIC -DPIC -o .libs/fs_lustre_file_open.o
In file included from /usr/include/linux/fs.h:19,
                 from /usr/include/linux/lustre/lustre_user.h:44,
                 from /usr/include/lustre/lustreapi.h:46,
                 from fs_lustre.h:37,
                 from fs_lustre_file_open.c:25:
/usr/include/sys/mount.h:35:3: error: expected identifier before numeric constant
   35 |   MS_RDONLY = 1,                /* Mount read-only.  */
      |   ^~~~~~~~~
```
This system is running 
```
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
```
Lustre version
```
hpp@login34:~/ompi> (main)cat /proc/fs/lustre/version
lustre: 2.15.0.7_rc2_cray_51_ga1c30e9
```

**Comment by hppritcha** [comment]
I should add i'm using gcc 11.2.0

**Comment by hppritcha** [comment]
we seem to be hitting a variant of https://github.com/systemd/systemd/issues/8507 and related.
I'm testing a PR to move inclusion of sys/mount.h back where it was before commit https://github.com/open-mpi/ompi/commit/c6a6c256e3687459a3c58bf31bff413a421b58a5

**Comment by edgargabriel** [comment]
> we seem to be hitting a variant of [systemd/systemd#8507](https://github.com/systemd/systemd/issues/8507) and related. I'm testing a PR to move inclusion of sys/mount.h back where it was before commit [c6a6c25](https://github.com/open-mpi/ompi/commit/c6a6c256e3687459a3c58bf31bff413a421b58a5)

@hppritcha thank you, lets see whether this fixes things. The commit does not really make major changes, it shifts however some header files from c-code to a header file instead. Not sure why that would trigger the failure, but its the only thing I can think of.

**Comment by edgargabriel** [comment]
@hppritcha I've added your pr to the v4.1 pull request https://github.com/open-mpi/ompi/pull/12171

**Comment by jsquyres** [comment]
v4.1.x PR has been merged: #12171 
(I didn't track what happened on main / v5.0.x)

**Comment by janjust** [comment]
fixed
